tests/helpers/tooling_contracts/workflows.py

```
@@ -91,6 +91,90 @@ def assert_ci_coverage_action_contract(ci_workflow: str) -> None:
+            f"expected generated audit workflow to include pinned {action}"
+        )
+    checkout_steps = [
+        step
+        for step in steps
+        if isinstance(step, dict) and step.get("uses") == checkout_action
+    ]
+    assert checkout_steps and all(
+        isinstance(step.get("with"), dict)
+        and step["with"].get("persist-credentials") is False
+        for step in checkout_steps
+    ), "expected generated audit workflow checkout to disable credential persistence"
```

Ruff (0.15.21)

[warning] Assertion should be broken down into multiple parts

(PT018)